pkg/frontend/adminactions/kubeactions.go

```diff
@@ -91,11 +92,18 @@ func (k *kubeActions) KubeList(ctx context.Context, groupKind, namespace string)
 		return nil, err
 	}
 
-	ul, err := k.dyn.Resource(*gvr).Namespace(namespace).List(ctx, metav1.ListOptions{})
+	// protect RP memory by not reading in more than 1000 items
+	ul, err := k.dyn.Resource(*gvr).Namespace(namespace).List(ctx, metav1.ListOptions{Limit: 1000})
 	if err != nil {
 		return nil, err
 	}
 
+	if ul.GetContinue() != "" {
+		return nil, api.NewCloudError(
+			http.StatusInternalServerError, api.CloudErrorCodeInternalServerError,
+			groupKind, "Too many items returned.")
+	}
+
```

Comment:
Should not we display a warning saying only the first 1000 items (provided the order of return is deterministic or can be made deterministic) are returned instead of an error and no result displayed ?